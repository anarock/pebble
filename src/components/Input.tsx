import * as React from "react";
import { cx } from "react-emotion";
import {
  InputProps,
  InputState,
  DateInputProps
} from "@src/components/typings/Input";
import Dropdown from "@src/components/Dropdown";
import Calendar from "@src/components/Calendar";
import format from "date-fns/format";
import {
  highlightStyle,
  labelStyle,
  wrapperStyle,
  messageStyle,
  inputStyle,
  dropDownClassName,
  dateClass
} from "@src/components/styles/Input.styles";
import { colors } from "@src/theme";

function getColor(error: string, success: string, isUnderlineColor?: boolean) {
  let color = colors.gray.dark;
  if (error) {
    color = colors.red.base;
  } else if (success) {
    color = colors.emerald.base;
  } else if (isUnderlineColor) {
    color = colors.violet.base;
  }

  return color;
}

class Input extends React.PureComponent<InputProps, InputState> {
  readonly state: InputState = {
    isFocused: false
  };

  static defaultProps = {
    value: "",
    readOnly: false
  };

  private addFocus = () => {
    this.setState({
      isFocused: true
    });
  };

  private removeFocus = () => {
    this.setState({
      isFocused: false
    });
  };

  private handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    this.props.onChange(e.target.value || "");
  };

  render() {
    const {
      type,
      placeholder,
      className,
      inputProps,
      fixLabelAtTop,
      value,
      readOnly,
      disabled,
      errorMessage,
      successMessage,
      message,
      textArea,
      required
    } = this.props;
    const { isFocused } = this.state;

    const _message = errorMessage || successMessage || message;

    const Input_ = textArea ? "textarea" : "input";

    const inputClassName = cx(inputStyle, {
      __pebble__input__disabled: disabled,
      __pebble__input__read__only: readOnly,
      __pebble__input__textarea: textArea
    });

    const highlightClassName = cx(highlightStyle, {
      __pebble__input__highlight__focused: isFocused,
      __pebble__input__highlight__state: !!_message,
      __pebble__input__highlight__read__only: readOnly,
      __pebble__input__highlight__disabled: disabled
    });

    const labelClassName = cx(labelStyle, {
      __pebble__input__label__focused: isFocused || !!value || fixLabelAtTop
    });

    const _wrapperStyle = cx(
      wrapperStyle,
      {
        __pebble__input__wrapper__textarea: textArea
      },
      className
    );

    return (
      <div
        className={_wrapperStyle}
        onFocus={this.addFocus}
        onBlur={this.removeFocus}
      >
        <label className={labelClassName}>
          {placeholder}
          {required && <span style={{ color: colors.red.base }}>&nbsp;*</span>}
        </label>

        <Input_
          className={inputClassName}
          {...inputProps}
          type={type}
          aria-label={placeholder}
          value={value || ""}
          onChange={this.handleChange}
          disabled={disabled}
          readOnly={readOnly}
        />

        <div
          className={highlightClassName}
          style={{
            backgroundColor: getColor(errorMessage, successMessage, true)
          }}
        />

        {_message && (
          <div
            className={messageStyle}
            style={{ color: getColor(errorMessage, successMessage) }}
          >
            {_message}
          </div>
        )}
      </div>
    );
  }
}

export default Input;

class DateInput extends React.PureComponent<DateInputProps> {
  onChange = (date: Date | string) => {
    this.props.onChange(date ? new Date(date).getTime() : undefined);
  };

  render() {
    const { value } = this.props;
    return (
      <Dropdown
        dropDownClassName={dropDownClassName}
        labelComponent={({ toggleDropdown }) => (
          <div onClick={toggleDropdown}>
            <Input
              onChange={this.onChange}
              type="date"
              value={format(value, "YYYY-MM-DD")}
              placeholder={this.props.placeholder}
              fixLabelAtTop
            />
          </div>
        )}
      >
        {({ toggle }) => (
          <Calendar
            hideShadow
            className={dateClass}
            selected={value ? new Date(value) : undefined}
            onChange={date => {
              this.onChange(date);
              toggle();
            }}
          />
        )}
      </Dropdown>
    );
  }
}

export { DateInput };
