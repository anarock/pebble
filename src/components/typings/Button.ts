import * as React from "react";

export type ButtonType = "primary" | "secondary" | "link" | "success" | "alert";

type Children = JSX.Element | string | number | React.ReactChildren;

export interface ButtonProps {
  children: Children;
  onClick: React.MouseEventHandler<HTMLButtonElement>;
  showShadow?: boolean;
  disabled?: boolean;
  type?: ButtonType;
  size?: "x-small" | "small" | "large";
  width?: number;
  className?: string;
  showRipple?: boolean;
  loading?: boolean;
  filled?: boolean;
}

export interface DropDownButtonProps extends ButtonProps {
  isOpen: boolean;
  isSelected: boolean;
}

interface MappingColorByTypeOptions {
  textColor?: string;
  base: string;
  hover: string;
  active: string;
  disabled: string;
}

export interface MappingColorByType {
  primary: MappingColorByTypeOptions;
  secondary: MappingColorByTypeOptions;
  success: MappingColorByTypeOptions;
  alert: MappingColorByTypeOptions;
}
