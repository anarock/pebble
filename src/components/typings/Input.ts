import * as React from "react";

interface CommonInputProps {
  placeholder: string;
  onChange: (text: string) => void;
  value?: string | number;
  className?: string;
  disabled?: boolean;
  errorMessage?: string;
  fixLabelAtTop?: boolean;
  inputClassName?: string;
  loading?: boolean;
  message?: string;
  onClick?: React.MouseEventHandler<HTMLDivElement>;
  readOnly?: boolean;
  required?: boolean;
  successMessage?: string;
}

interface SimpleInputProps extends CommonInputProps {
  inputProps?: React.InputHTMLAttributes<HTMLInputElement>;
  textArea?: false;
  type?: "text" | "date" | "password" | "number" | "email";
}

interface TextAreaInputProps extends CommonInputProps {
  inputProps?: React.InputHTMLAttributes<HTMLTextAreaElement>;
  textArea: true;
  type?: undefined;
}

export type InputProps = SimpleInputProps | TextAreaInputProps;

export interface InputState {
  isFocused: boolean;
}
