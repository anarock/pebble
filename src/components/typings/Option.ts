import { ControlProps } from "./Control";
import { Omit } from "utility-types";
import * as React from "react";

export interface OptionProps<OptionType>
  extends Omit<ControlProps<OptionType>, "checked" | "type"> {
  isActive?: boolean;
  isSelected?: boolean;
  multiSelect?: boolean;
  labelClassName?: string;
  rightElement?: (props: OptionProps<OptionType>) => React.ReactNode;
}
