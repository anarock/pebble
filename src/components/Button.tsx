import * as React from "react";
import { cx } from "emotion";
import {
  iconStyle,
  getButtonStyle,
  dropDownButtonStyle,
  dropDownButtonDefaultStyle
} from "./styles/Button.styles";
import Ink from "react-ink";
import { ButtonProps, DropDownButtonProps } from "./typings/Button";
import Loader from "./Loader";
import { colors } from "../theme";

const Button: React.FunctionComponent<ButtonProps> = ({
  type = "primary",
  disabled,
  children,
  onClick,
  width,
  showShadow,
  className,
  showRipple = true,
  loading,
  size = "small"
}: ButtonProps) => {
  const disableAction = disabled || loading;

  const filled = size !== "x-small";
  const _className = cx(
    getButtonStyle(size, type, !!showShadow, filled),
    className
  );

  return (
    <button
      className={_className}
      onClick={!disableAction ? onClick : undefined}
      style={{ width }}
      disabled={disabled}
    >
      {loading ? <Loader color={colors.white.base} scale={0.4} /> : children}
      {!disableAction && showRipple && type !== "link" && <Ink />}
    </button>
  );
};

export const DropDownButton: React.FunctionComponent<DropDownButtonProps> = ({
  isOpen,
  isSelected,
  children,
  className,
  ...props
}) => {
  const _className = cx(dropDownButtonStyle, {
    [dropDownButtonDefaultStyle]: !(isOpen || isSelected)
  });

  return (
    <Button {...props} type="secondary" className={cx(_className, className)}>
      <React.Fragment>
        {children}{" "}
        <i
          className={cx("pi pi-arrow-drop-down", iconStyle)}
          style={{
            transform: isOpen ? "rotate(180deg)" : "none"
          }}
        />
      </React.Fragment>
    </Button>
  );
};

export default Button;
