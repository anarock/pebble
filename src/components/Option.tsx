import * as React from "react";
import { OptionProps } from "./typings/Option";
import Control from "./shared/Control";
import { cx } from "emotion";
import { activeRow, rowWrapper, selectedRow } from "./styles/Options.styles";
import Ink from "react-ink";
import { colors, mixins } from "../theme";
import { Component } from "react";

class Option extends Component<OptionProps> {
  static defaultProps: Partial<OptionProps> = {
    rightElement: ({ isSelected, multiSelect }) => {
      const iconClass = cx({
        "icon-checkbox-selected": isSelected,
        "icon-checkbox-unselected": !isSelected
      });
      return multiSelect ? (
        <i
          style={{
            color: isSelected ? colors.violet.base : colors.gray.light
          }}
          className={iconClass}
        />
      ) : null;
    }
  };

  render() {
    return (
      <Control
        {...this.props}
        checked={this.props.isSelected}
        type={this.props.multiSelect ? "checkbox" : "radio"}
      >
        {() => {
          const { label, isActive, isSelected, rightElement } = this.props;

          const _class = cx(rowWrapper, {
            [activeRow]: isActive,
            [selectedRow]: isSelected
          });

          return (
            <div className={_class} style={mixins.flexSpaceBetween}>
              {label} {rightElement(this.props)}
              <Ink />
            </div>
          );
        }}
      </Control>
    );
  }
}

export default Option;
