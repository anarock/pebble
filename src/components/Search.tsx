import * as React from "react";
import { cx } from "emotion";
import { SearchProps, SearchState } from "./typings/Search";
import {
  searchStyle,
  searchWrapperStyle,
  clearContainer
} from "./styles/Search.styles";

class Search extends React.PureComponent<SearchProps, SearchState> {
  searchInputRef: React.RefObject<HTMLInputElement> = React.createRef();

  static defaultProps: Partial<SearchProps> = {
    showSearchIcon: true,
    clearable: true,
    onClear: () => {}
  };

  state: SearchState = {
    searchValue: ""
  };

  render() {
    const {
      type,
      onChange,
      placeholder,
      showSearchIcon,
      className,
      clearable,
      onClear
    } = this.props;
    const { searchValue } = this.state;

    const wrapperClassName = cx(searchWrapperStyle, {
      __pebble__search__small: type === "small",
      __pebble__search__large: type === "large",
      __pebble__search__table: type === "table"
    });

    return (
      <div className={cx(wrapperClassName, className)}>
        {type !== "large" && showSearchIcon && <i className="pi pi-search" />}
        <input
          className={searchStyle}
          type="text"
          aria-label={placeholder}
          placeholder={placeholder}
          onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
            this.setState({ searchValue: e.target.value });
            onChange(e.target.value);
          }}
          ref={this.searchInputRef}
        />
        {clearable && (
          <div
            className={cx(clearContainer, { __display: !!searchValue.length })}
            onClick={() => {
              this.setState({ searchValue: "" });
              if (this.searchInputRef.current) {
                this.searchInputRef.current.value = "";
              }
              onClear();
            }}
          >
            <i
              className="pi pi-close"
              style={{ display: "table-cell", verticalAlign: "middle" }}
            />
          </div>
        )}
      </div>
    );
  }
}

export default Search;
