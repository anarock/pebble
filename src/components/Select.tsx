import * as React from "react";
import { cx } from "emotion";
import { SelectProps } from "./typings/Select";
import {
  chevronStyle,
  dropDownClass,
  inputWrapper,
  selectInput,
  selectInputWrapper,
  selectWrapper,
  fullWidth,
  relativePosition
} from "./styles/Select.styles";
import DropDown from "./DropDown";
import Input from "./Input";
import OptionGroupCheckBox from "./OptionGroupCheckBox";
import OptionGroupRadio from "./OptionGroupRadio";

function noop() {}

const Select: React.FunctionComponent<SelectProps> = props => {
  const {
    className,
    placeholder,
    required,
    errorMessage,
    value,
    dropdownClassName,
    inputProps,
    fullWidthDropdown,
    onDropdownToggle = noop,
    disabled
  } = props;

  return (
    <div
      className={cx(selectWrapper, className, {
        [relativePosition]: fullWidthDropdown
      })}
    >
      <DropDown
        dropDownClassName={cx(dropDownClass, dropdownClassName, {
          [fullWidth]: fullWidthDropdown
        })}
        onOutsideClick={isOpen => onDropdownToggle(isOpen)}
        labelComponent={({ toggleDropdown, isOpen }) => {
          const chevron = cx(chevronStyle, "pi", "pi-arrow-drop-down", {
            __pebble__select__open: isOpen
          });
          return (
            <div
              className={inputWrapper}
              onClick={
                disabled
                  ? undefined
                  : () => {
                      toggleDropdown();
                      onDropdownToggle(isOpen);
                    }
              }
            >
              <Input
                className={selectInputWrapper}
                inputClassName={selectInput}
                placeholder={placeholder}
                value={value || ""}
                onChange={noop}
                required={required}
                message={isOpen ? " " : ""}
                errorMessage={errorMessage}
                readOnly
                disabled={disabled}
                {...inputProps}
              />
              <i className={chevron} />
            </div>
          );
        }}
      >
        {({ toggle, isOpen }) => {
          const { children, onClear, searchBox, searchBoxProps } = props;
          const commonProps = {
            onClear:
              onClear &&
              (() => {
                onClear();
                onDropdownToggle(isOpen);
                toggle();
              }),
            searchBox,
            searchBoxProps
          };

          // This would have been the ideal way to write this but tyepscript is crying.
          // const OptionGroup = props.multiSelect ? OptionGroupCheckBox : OptionGroupRadio;
          // return (
          //   <OptionGroup
          //     selected={props.selected}
          //     onChange={(_value, extras) => {
          //       props.onChange(_value, extras)
          //       props.multiSelect && toggle();
          //     }}
          //     onApply={props.multiSelect && ((_value) => {
          //       props.onApply && props.onApply(_value, props);
          //       toggle();
          //     })}
          //     {...commonProps}
          //   >
          //     {children}
          //   </OptionGroup>
          // )

          if (props.multiSelect === true) {
            return (
              <OptionGroupCheckBox
                selected={props.selected}
                onChange={(_value, extras) => {
                  props.onChange(_value, extras);
                }}
                onApply={
                  props.onApply &&
                  (_value => {
                    if (props.onApply) props.onApply(_value, props);
                    onDropdownToggle(isOpen);
                    toggle();
                  })
                }
                {...commonProps}
              >
                {children}
              </OptionGroupCheckBox>
            );
          } else {
            return (
              <OptionGroupRadio
                selected={props.selected}
                onChange={(_value, extras) => {
                  if (_value) props.onChange(_value, extras);
                  onDropdownToggle(isOpen);
                  toggle();
                }}
                {...commonProps}
              >
                {children}
              </OptionGroupRadio>
            );
          }
        }}
      </DropDown>
    </div>
  );
};

export default Select;
