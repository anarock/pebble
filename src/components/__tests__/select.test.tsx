import * as React from "react";
import renderer from "react-test-renderer";
import Option from "../Option";
import Select from "../Select";
import { mount } from "enzyme";
import sinon from "sinon";
import { SingleSelectProps, MultiSelectProps } from "../typings/Select";
import Button from "../Button";
import Input from "../Input";
import Search from "../Search";

const options = new Array(5)
  .fill(1)
  .map((_x, i) => (
    <Option key={i + 1} value={`option-${i + 1}`} label="I am an option" />
  ));

const noop = () => {};

function getComponent(spy = noop, props: Partial<SingleSelectProps> = {}) {
  return (
    <Select
      onChange={spy}
      placeholder="Choose Option"
      selected={"option-2"}
      {...props}
    >
      {options}
    </Select>
  );
}

function getMultiSelectComponent(
  spy = noop,
  props: Partial<MultiSelectProps> = {}
) {
  return (
    <Select onChange={spy} placeholder="Choose Option" multiSelect {...props}>
      {options}
    </Select>
  );
}

describe("Component: Select", () => {
  test("single-select: snapshot", () => {
    const select = renderer.create(getComponent());
    const tree = select.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("multi-select: snapshot", () => {
    const select = renderer.create(
      getMultiSelectComponent(undefined, {
        multiSelect: true,
        selected: []
      })
    );
    const tree = select.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("single-select with searchbox: snapshot", () => {
    const select = renderer.create(
      getComponent(undefined, {
        searchBox: true,
        searchBoxProps: {
          placeholder: "Search"
        }
      })
    );
    const tree = select.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should open when input is clicked", () => {
    const spy = sinon.spy();
    const select = mount(getComponent(spy));

    expect(select.find(Option)).toHaveLength(0);

    select.find(Input).simulate("click");
    expect(select.find(Option)).toHaveLength(5);
  });

  test("single select: should trigger onChange with correct onChange", () => {
    const spy = sinon.spy();
    const select = mount(getComponent(spy));
    select.find(Input).simulate("click");
    select
      .find(Option)
      .at(2)
      .simulate("click");

    expect(spy.calledWith("option-3")).toBeTruthy();
    expect(select.find(Option)).toHaveLength(0);
  });

  test("multi select: should trigger onChange with correct onChange", () => {
    const spy = sinon.spy();
    const applySpy = sinon.spy();
    const clearSpy = sinon.spy();
    const select = mount(
      getMultiSelectComponent(spy, {
        multiSelect: true,
        onApply: applySpy,
        onClear: clearSpy,
        selected: []
      })
    );
    select.find(Input).simulate("click");
    select
      .find(Option)
      .at(2)
      .simulate("click");

    expect(spy.calledWith(["option-3"])).toBeTruthy();

    select.setProps({
      selected: ["option-3"]
    });

    select
      .find(Option)
      .at(3)
      .simulate("click");

    expect(spy.calledWith(["option-3", "option-4"])).toBeTruthy();

    select.setProps({
      selected: ["option-3", "option-4"]
    });

    select
      .find(Option)
      .at(2)
      .simulate("click");

    expect(spy.calledWith(["option-4"])).toBeTruthy();

    expect(select.find(Button)).toHaveLength(2);

    // test onClear
    select
      .find(Button)
      .at(0)
      .simulate("click");
    expect(clearSpy.calledOnce).toBeTruthy();

    // Reopen dropdown and test onApply
    select.find(Input).simulate("click");
    select.setProps({
      selected: ["option-3"]
    });

    select
      .find(Button)
      .at(1)
      .simulate("click");
    expect(applySpy.calledWith(["option-3"])).toBeTruthy();

    // ensure the dropdown is closed
    expect(select.find(Option)).toHaveLength(0);
  });

  test("single select: should trigger onChange with correct onChange", () => {
    const spy = sinon.spy();
    const queryChangeSpy = sinon.spy();
    const select = mount(
      getComponent(spy, {
        searchBox: true,
        searchBoxProps: {
          placeholder: "Search",
          onChange: queryChangeSpy
        }
      })
    );
    select.find(Input).simulate("click");
    select
      .find(Search)
      .find("input")
      .simulate("change", {
        target: {
          value: "hello"
        }
      });

    expect(queryChangeSpy.calledWith("hello")).toBeTruthy();
  });
});
