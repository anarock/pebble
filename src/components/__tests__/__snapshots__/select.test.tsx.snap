// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Select multi-select: snapshot 1`] = `
.emotion-6 {
  margin-bottom: 20px;
}

.emotion-5 {
  cursor: pointer;
  position: relative;
}

.emotion-4 {
  position: absolute;
  top: 25px;
  right: 7px;
  color: #9FA6B0;
  font-size: 10px;
}

.emotion-4.__pebble__select__open {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.emotion-1 {
  outline: 0;
  border: 0;
  border-bottom: 1px solid #EDEDED;
  padding: 24px 0 12px 0;
  height: 48px;
  border-radius: 0;
  font-weight: 400;
  color: #101721;
  font-size: 14px;
  width: 100%;
  overflow-x: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.emotion-1:disabled backgroundColor {
  base: #ffffff;
}

.emotion-1._pebble_input_read_only {
  color: #6B7785;
}

.emotion-1._pebble_input_disabled {
  cursor: not-allowed;
  pointer-events: none;
  color: #9FA6B0;
}

.emotion-1._pebble_input_textarea {
  height: 88px;
  padding: 24px 0 52px 0;
  resize: none;
}

.emotion-1[type='date'] {
  -webkit-appearance: textfield;
}

.emotion-1[type='date']::-webkit-inner-spin-button,
.emotion-1[type='date']::-webkit-calendar-picker-indicator {
  webkit-appearance: none;
  display: none;
}

.emotion-1::-webkit-input-placeholder {
  color: #E0E0E0;
}

.emotion-1::-moz-placeholder {
  color: #E0E0E0;
}

.emotion-1:-ms-input-placeholder {
  color: #E0E0E0;
}

.emotion-1:-moz-placeholder {
  color: #E0E0E0;
}

.emotion-1::-webkit-input-placeholder {
  color: #E0E0E0;
}

.emotion-1::-moz-placeholder {
  color: #E0E0E0;
}

.emotion-1:-ms-input-placeholder {
  color: #E0E0E0;
}

.emotion-1::placeholder {
  color: #E0E0E0;
}

.emotion-2 {
  height: 2px;
  background-color: #6161FF;
  margin-top: -2px;
  position: relative;
  width: 0;
  -webkit-transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  z-index: 9;
}

.emotion-2._pebble_input_highlight_state,
.emotion-2._pebble_input_highlight_focused {
  width: 100%;
}

.emotion-0 {
  color: #9FA6B0;
  font-size: 14px;
  line-height: 12px;
  position: absolute;
  -webkit-transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  -webkit-transform: translate(0,24px);
  -ms-transform: translate(0,24px);
  transform: translate(0,24px);
  pointer-events: none;
}

.emotion-0._pebble_input_label_focused {
  font-size: 12px;
  line-height: 10px;
  -webkit-transform: translate(0,0);
  -ms-transform: translate(0,0);
  transform: translate(0,0);
  color: #6B7785;
}

.emotion-3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #ffffff;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 20px;
  height: 68px;
  pointer-events: none;
}

.emotion-3._pebble_input_wrapper_textarea {
  height: 110px;
}

<div
  className="emotion-6"
>
  <div>
    <div>
      <div
        className="emotion-5"
        onClick={[Function]}
      >
        <div
          className="emotion-3"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <label
            className="emotion-0"
          >
            Choose Option
          </label>
          <input
            aria-label="Choose Option"
            className="emotion-1 _pebble_input_read_only"
            onChange={[Function]}
            readOnly={true}
            value=""
          />
          <div
            className="emotion-2 _pebble_input_highlight_read_only"
            style={
              Object {
                "backgroundColor": "#6161FF",
              }
            }
          />
          
        </div>
        <i
          className="emotion-4 icon-arrow-drop-down"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Component: Select single-select: snapshot 1`] = `
.emotion-6 {
  margin-bottom: 20px;
}

.emotion-5 {
  cursor: pointer;
  position: relative;
}

.emotion-4 {
  position: absolute;
  top: 25px;
  right: 7px;
  color: #9FA6B0;
  font-size: 10px;
}

.emotion-4.__pebble__select__open {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.emotion-1 {
  outline: 0;
  border: 0;
  border-bottom: 1px solid #EDEDED;
  padding: 24px 0 12px 0;
  height: 48px;
  border-radius: 0;
  font-weight: 400;
  color: #101721;
  font-size: 14px;
  width: 100%;
  overflow-x: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.emotion-1:disabled backgroundColor {
  base: #ffffff;
}

.emotion-1._pebble_input_read_only {
  color: #6B7785;
}

.emotion-1._pebble_input_disabled {
  cursor: not-allowed;
  pointer-events: none;
  color: #9FA6B0;
}

.emotion-1._pebble_input_textarea {
  height: 88px;
  padding: 24px 0 52px 0;
  resize: none;
}

.emotion-1[type='date'] {
  -webkit-appearance: textfield;
}

.emotion-1[type='date']::-webkit-inner-spin-button,
.emotion-1[type='date']::-webkit-calendar-picker-indicator {
  webkit-appearance: none;
  display: none;
}

.emotion-1::-webkit-input-placeholder {
  color: #E0E0E0;
}

.emotion-1::-moz-placeholder {
  color: #E0E0E0;
}

.emotion-1:-ms-input-placeholder {
  color: #E0E0E0;
}

.emotion-1:-moz-placeholder {
  color: #E0E0E0;
}

.emotion-1::-webkit-input-placeholder {
  color: #E0E0E0;
}

.emotion-1::-moz-placeholder {
  color: #E0E0E0;
}

.emotion-1:-ms-input-placeholder {
  color: #E0E0E0;
}

.emotion-1::placeholder {
  color: #E0E0E0;
}

.emotion-2 {
  height: 2px;
  background-color: #6161FF;
  margin-top: -2px;
  position: relative;
  width: 0;
  -webkit-transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  z-index: 9;
}

.emotion-2._pebble_input_highlight_state,
.emotion-2._pebble_input_highlight_focused {
  width: 100%;
}

.emotion-0 {
  color: #9FA6B0;
  font-size: 14px;
  line-height: 12px;
  position: absolute;
  -webkit-transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  -webkit-transform: translate(0,24px);
  -ms-transform: translate(0,24px);
  transform: translate(0,24px);
  pointer-events: none;
}

.emotion-0._pebble_input_label_focused {
  font-size: 12px;
  line-height: 10px;
  -webkit-transform: translate(0,0);
  -ms-transform: translate(0,0);
  transform: translate(0,0);
  color: #6B7785;
}

.emotion-3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #ffffff;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 20px;
  height: 68px;
  pointer-events: none;
}

.emotion-3._pebble_input_wrapper_textarea {
  height: 110px;
}

<div
  className="emotion-6"
>
  <div>
    <div>
      <div
        className="emotion-5"
        onClick={[Function]}
      >
        <div
          className="emotion-3"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <label
            className="emotion-0"
          >
            Choose Option
          </label>
          <input
            aria-label="Choose Option"
            className="emotion-1 _pebble_input_read_only"
            onChange={[Function]}
            readOnly={true}
            value=""
          />
          <div
            className="emotion-2 _pebble_input_highlight_read_only"
            style={
              Object {
                "backgroundColor": "#6161FF",
              }
            }
          />
          
        </div>
        <i
          className="emotion-4 icon-arrow-drop-down"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Component: Select sinle-select with searchbox: snapshot 1`] = `
.emotion-6 {
  margin-bottom: 20px;
}

.emotion-5 {
  cursor: pointer;
  position: relative;
}

.emotion-4 {
  position: absolute;
  top: 25px;
  right: 7px;
  color: #9FA6B0;
  font-size: 10px;
}

.emotion-4.__pebble__select__open {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.emotion-1 {
  outline: 0;
  border: 0;
  border-bottom: 1px solid #EDEDED;
  padding: 24px 0 12px 0;
  height: 48px;
  border-radius: 0;
  font-weight: 400;
  color: #101721;
  font-size: 14px;
  width: 100%;
  overflow-x: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.emotion-1:disabled backgroundColor {
  base: #ffffff;
}

.emotion-1._pebble_input_read_only {
  color: #6B7785;
}

.emotion-1._pebble_input_disabled {
  cursor: not-allowed;
  pointer-events: none;
  color: #9FA6B0;
}

.emotion-1._pebble_input_textarea {
  height: 88px;
  padding: 24px 0 52px 0;
  resize: none;
}

.emotion-1[type='date'] {
  -webkit-appearance: textfield;
}

.emotion-1[type='date']::-webkit-inner-spin-button,
.emotion-1[type='date']::-webkit-calendar-picker-indicator {
  webkit-appearance: none;
  display: none;
}

.emotion-1::-webkit-input-placeholder {
  color: #E0E0E0;
}

.emotion-1::-moz-placeholder {
  color: #E0E0E0;
}

.emotion-1:-ms-input-placeholder {
  color: #E0E0E0;
}

.emotion-1:-moz-placeholder {
  color: #E0E0E0;
}

.emotion-1::-webkit-input-placeholder {
  color: #E0E0E0;
}

.emotion-1::-moz-placeholder {
  color: #E0E0E0;
}

.emotion-1:-ms-input-placeholder {
  color: #E0E0E0;
}

.emotion-1::placeholder {
  color: #E0E0E0;
}

.emotion-2 {
  height: 2px;
  background-color: #6161FF;
  margin-top: -2px;
  position: relative;
  width: 0;
  -webkit-transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  z-index: 9;
}

.emotion-2._pebble_input_highlight_state,
.emotion-2._pebble_input_highlight_focused {
  width: 100%;
}

.emotion-0 {
  color: #9FA6B0;
  font-size: 14px;
  line-height: 12px;
  position: absolute;
  -webkit-transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  transition: all 0.3s cubic-bezier(.64,.09,.08,1);
  -webkit-transform: translate(0,24px);
  -ms-transform: translate(0,24px);
  transform: translate(0,24px);
  pointer-events: none;
}

.emotion-0._pebble_input_label_focused {
  font-size: 12px;
  line-height: 10px;
  -webkit-transform: translate(0,0);
  -ms-transform: translate(0,0);
  transform: translate(0,0);
  color: #6B7785;
}

.emotion-3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #ffffff;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 20px;
  height: 68px;
  pointer-events: none;
}

.emotion-3._pebble_input_wrapper_textarea {
  height: 110px;
}

<div
  className="emotion-6"
>
  <div>
    <div>
      <div
        className="emotion-5"
        onClick={[Function]}
      >
        <div
          className="emotion-3"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <label
            className="emotion-0"
          >
            Choose Option
          </label>
          <input
            aria-label="Choose Option"
            className="emotion-1 _pebble_input_read_only"
            onChange={[Function]}
            readOnly={true}
            value=""
          />
          <div
            className="emotion-2 _pebble_input_highlight_read_only"
            style={
              Object {
                "backgroundColor": "#6161FF",
              }
            }
          />
          
        </div>
        <i
          className="emotion-4 icon-arrow-drop-down"
        />
      </div>
    </div>
  </div>
</div>
`;
